src/libraries/System.Security.Cryptography/tests/HashAlgorithmTest.cs

```diff
@@ -123,15 +130,19 @@ protected override void HashCore(byte[] array, int ibStart, int cbSize)
             // implementations by verifying the right value is produced.
         }
 
-        private class SlowPositionValueStream : PositionValueStream
+        private class SelfCancellingStream : PositionValueStream
         {
-            public SlowPositionValueStream(int totalCount) : base(totalCount)
+            private readonly CancellationTokenSource _cancellationSource;
+
+            public SelfCancellingStream(int totalCount, CancellationTokenSource cancellationSource)
+                : base(totalCount)
             {
+                _cancellationSource = cancellationSource;
             }
 
             public override int Read(byte[] buffer, int offset, int count)
             {
-                System.Threading.Thread.Sleep(1000);
+                _cancellationSource.Cancel(throwOnFirstException: true);
```

Comment:
What operation is being canceled here, what state is it in when canceled, and what is the test trying to exercise?  I'm wondering if this is actually negating the test by issuing cancellation before interesting stuff happens, whereas previously it might have been while interesting stuff happens.

Reply:
Ok, then seems reasonable.